src/Sql/Sql.Test/ScenarioTests/InstancePoolCrudTests.ps1

```diff
@@ -0,0 +1,452 @@
+﻿# ----------------------------------------------------------------------------------
+#
+# Copyright Microsoft Corporation
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+# http://www.apache.org/licenses/LICENSE-2.0
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+# ----------------------------------------------------------------------------------
+
+
+# ----------------------------------------------------------------------------------
+# Instance pool level tests
+# For the following tests below, see Common.ps1 to identify helpers (e.g. Create-InstancePoolForTest)
+# ----------------------------------------------------------------------------------
+
+<#
+    .SYNOPSIS
+    Tests creating an instance pool
+#>
+function Test-CreateInstancePool
+{
+    # Create an instance pool
+    $props = Get-InstancePoolTestProperties
+    $instancePool = New-AzSqlInstancePool -ResourceGroupName $props.resourceGroup -Name $props.name `
+                    -Location $props.location -SubnetId $props.SubnetId -VCore $props.vCores `
+                    -Edition $props.edition -ComputeGeneration $props.computeGen `
+                    -LicenseType $props.licenseType -Tag $props.tags
+
+    Assert-InstancePoolProperties $instancePool
+
+    # Clean up instances in pool
+    Remove-ManagedInstancesInInstancePool($instancePool)
+}
+
+<#
+    .SYNOPSIS
+    Tests getting an instance pool
+#>
+function Test-GetInstancePool
+{
+    # Setup
+    $instancePool = Create-InstancePoolForTest
+    Assert-InstancePoolProperties $instancePool
+
+    # Test get single - default params
+    $instancePool = Get-AzSqlInstancePool -ResourceGroupName $instancePool.ResourceGroupName -Name $instancePool.InstancePoolName
```

Comment:
@johnpaulkee a quick refactoring comment: for tests where you're making multiple cmdlet calls, I would advise to follow a pattern similar to the following:

```powershell
# Create necessary resources for test before try-finally
try
{
    # Call cmdlets to test and verify behavior with assertions
}
finally
{
    # Delete resource(s) and/or resource group(s) created during this test
}
```

With this model, you're guaranteed to delete all resources when recording the test regardless of if there's an exception thrown or not.